File: Bubble_Sort/Bubble_Sort.java
package Bubble_Sort;

public class Bubble_Sort {
    /**
     * Sorts an array of integers using the Bubble Sort algorithm.
     * 
     * @param arr The array to be sorted.
     */
    public static void sort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // Swap elements if they are in the wrong order
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    /**
     * Main function demonstrating the usage of the Bubble Sort algorithm.
     * 
     * @param args Command-line arguments (not used in this case).
     */
    public static void main(String[] args) {
        int[] arr = { 64, 25, 12, 22, 11 };

        // Print the original list
        System.out.print("Original List:\t");
        for (int i : arr) {
            System.out.print(i + " ");
        }
        System.out.println();

        // Call the sort function to sort the array
        sort(arr);

        // Print the sorted list
        System.out.print("Sorted List:\t");
        for (int i : arr) {
            System.out.print(i + " ");
        }
        System.out.println();
    }
}

File: Bubble_Sort/Bubble_Sort.go
package main

import "fmt"

// BubbleSort sorts a slice of numbers using the Bubble Sort algorithm.
func BubbleSort(arr []int) []int {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		for j := 0; j < n-i-1; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}

	return arr
}

func main() {
	// Example usage
	arr := []int{64, 25, 12, 22, 11}

	// Print the original list
	fmt.Print("Original List: ")
	fmt.Println(arr)

	// Call the BubbleSort function to sort the slice
	arr = BubbleSort(arr)

	// Print the sorted list
	fmt.Print("Sorted List: ")
	fmt.Println(arr)
}
